bionic/datatypes.py

```diff
@@ -72,15 +69,50 @@ class Result(object):
     Represents one value for one entity.
     '''
     def __init__(
-            self, query, value, file_path=None):
+            self, query, value, value_hash=None, file_path=None):
         self.query = query
         self.value = value
+        # Only present when value should persist.
         self.file_path = file_path
+        self.value_hash = value_hash
+
+    def to_hashed_result(self):
+        return HashedResult(self.query, self.value_hash)
 
     def __repr__(self):
         return f'Result({self.query!r}, {self.value!r})'
 
 
+class HashedResult(object):
```

Comment:
So, given the way the code looks now: does this need to be a class? Could we just pass around the value hashes themselves?

Reply:
Well, that's how I started it but had to make it a class. I see no reason to keep the class now with the refactor since I can get the queries (for map key name) from other entities loading the hash and creating the map.